src/app/mercado/criar/page.tsx

```
@@ -0,0 +1,75 @@
+  const [imageUrl, setImageUrl] = useState('');
+  const [error, setError] = useState<string | null>(null);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    try {
+      const { error: err } = await supabase
+        .from('marketplace')
+        .insert([{ title, description, price: parseFloat(price), image_url: imageUrl || null }]);
+      if (err) throw new Error(err.message);
+      router.push('/mercado');
+    } catch (err: any) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)